app/uk/gov/hmrc/hecstubs/controllers/AccountingPeriodController.scala

```diff
@@ -0,0 +1,111 @@
+/*
+ * Copyright 2021 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.hecstubs.controllers
+
+import cats.data.Validated.{Invalid, Valid}
+import cats.data.{Validated, ValidatedNel}
+import cats.syntax.apply._
+import com.google.inject.{Inject, Singleton}
+import play.api.libs.json.Json
+import play.api.mvc.{Action, AnyContent, ControllerComponents}
+import uk.gov.hmrc.hecstubs.models.accountOverviewDetails.ErrorCode._
+import uk.gov.hmrc.hecstubs.models.accountOverviewDetails.ReturnStatus.ReturnFound
+import uk.gov.hmrc.hecstubs.models.accountOverviewDetails.{Environment, ErrorResponse, ErrorResult}
+import uk.gov.hmrc.hecstubs.models.companyAccountingPeriod.{AccountingPeriod, CTUTR, CompanyAccountingPeriod}
+import uk.gov.hmrc.hecstubs.util.{Logging, ValidationUtils}
+import uk.gov.hmrc.play.bootstrap.backend.controller.BackendController
+
+import java.time.LocalDate
+import java.util.UUID
+
+@Singleton
+class AccountingPeriodController @Inject() (cc: ControllerComponents) extends BackendController(cc) with Logging {
+
+  val message    = "Submission has not passed validation."
```

Comment:
small thing here: `message` seems quite vague as a name - perhaps something like `badRequestMessage` or `validationErrorMessage` is a bit more descriptive. There's a similar thing in `IndividualAccountOverviewController`

Reply:
updated